python/dolma/cli/deduper.py

```diff
@@ -148,6 +156,13 @@ def run(cls, parsed_config: DeduperConfig):
                 try_name = try_name or cfg["attribute_name"]
             else:
                 raise ValueError("Either dedupe.documents or dedupe.paragraphs must be specified")
+            if (
+                dedupe_dict_config["num_partitions"] > 1
+                and parsed_config.dedupe.paragraphs
+                and parsed_config.dedupe.paragraphs.by_ngram
+                and (parsed_config.dedupe.paragraphs.by_ngram.ngram_length or 0) > 0
+            ):
+                raise ValueError("Work partitioning is not available for n-gram deduplication")
```

Comment:
I think we want this partitioning to work also for ngrams, not just full documents.

Reply:
Because of the additional logic around ngram-based paragraph deduplication, this would require an additional downstream step to make work. 

For document level dedupe, it's basically the same as current - one doc gets filtered to one host, gets written to a file as duplicate or not.

For paragraph level dedupe it's a little messier - one doc's paragraphs will be looked at by multiple hosts and information about that doc will be stored in multiple files. That's fine though, they're in the same S3 folder and can be collated easily.

For n-gram level dedupe, each host would have a bloom filter containing info for a fraction of n-grams across all documents, which would mean that a single host couldn't determine if a paragraph was a duplicate or not, it would need to either perform some r/w to a remote store to keep track of those 0-1 values and indicate it's a duplicate once that value hits 1, or a downstream process would have to collate the  files generated by each host and then go paragraph by paragraph and sum up a score to determine if it's a duplicate or not. Either way I feel like there's a good chance that added complexity would negate the value of doing partitioning.